reposerver/repository/repository.go

```diff
@@ -229,14 +231,35 @@ func helmTemplate(appPath string, env *v1alpha1.Env, q *apiclient.ManifestReques
 		Set:         map[string]string{},
 		SetString:   map[string]string{},
 	}
+
+	baseDirectoryPath := security.SubtractRelativeFromAbsolutePath(appPath, q.ApplicationSource.Path)
 	appHelm := q.ApplicationSource.Helm
 	if appHelm != nil {
 		if appHelm.ReleaseName != "" {
 			templateOpts.Name = appHelm.ReleaseName
 		}
-		templateOpts.Values = appHelm.ValueFiles
+
+		for _, val := range appHelm.ValueFiles {
+			absBaseDir, err := filepath.Abs(baseDirectoryPath)
+			if err != nil {
+				return nil, err
+			}
+			if !filepath.IsAbs(val) {
+				absWorkDir, err := filepath.Abs(appPath)
+				if err != nil {
+					return nil, err
+				}
+				val = filepath.Join(absWorkDir, val)
+			}
+			cleanVal, err := security.EnforceToCurrentRoot(absBaseDir, val)
+			if err != nil {
+				return nil, err
+			}
+			templateOpts.Values = append(templateOpts.Values, cleanVal)
```

Comment:
what do we need cleanVal for? can't we use the original?

Reply:
`cleanVal` is the absolute path of the original file. We don't _need_ it, but since we use it for the directory enforcer I elected to use it. Can remove if necessary

Reply:
I suspect you'll find it easier to deal with URLs if you remove it. Not mandatory.

Reply:
You were right! Using `val` now